receiver/oracledbreceiver/scraper.go

```diff
@@ -828,3 +828,43 @@ func (s *oracleScraper) shutdown(_ context.Context) error {
 	}
 	return s.db.Close()
 }
+
+func (s *oracleScraper) setupResourceBuilder(rb *metadata.ResourceBuilder) *metadata.ResourceBuilder {
+	rb.SetOracledbInstanceName(s.instanceName)
+	rb.SetHostName(s.hostName)
+	rb.SetServiceInstanceID(s.serviceInstanceID)
+	return rb
+}
+
+func getInstanceID(instanceString string, logger *zap.Logger) string {
+	parts := strings.SplitN(instanceString, "/", 2)
+	hostAndPort := parts[0]
+	service := ""
+	if len(parts) == 2 {
+		service = "/" + parts[1]
+	}
+
+	host, port, err := net.SplitHostPort(hostAndPort)
+	if err != nil {
+		return handleInstanceIDErr(err, logger, port, service)
+	}
+
+	if strings.EqualFold(host, "localhost") || net.ParseIP(host).IsLoopback() {
+		host, err = os.Hostname()
+	}
+	if err != nil {
+		return handleInstanceIDErr(err, logger, port, service)
+	}
+
+	return host + ":" + port + service
+}
+
+func handleInstanceIDErr(err error, logger *zap.Logger, port, service string) string {
+	const fallback = "unknown:1521"
```

Comment:
It would be good to document default value in the error case, so if users see `unknown:1521` as the `service.instance.id` they know why it's being set to that value, and what they can do (if anything) to get it to be a useful value.